subsys/net/lib/mqtt_helper/mqtt_helper.c

```diff
@@ -367,9 +367,25 @@ static int broker_init(struct sockaddr_storage *broker, struct mqtt_helper_conn_
 	}
 
 	addr = result;
+	bool found = false;
 
 	while (addr != NULL) {
-		if (addr->ai_addrlen == sizeof(struct sockaddr_in)) {
+		if (addr->ai_family == AF_INET6) {
+			struct sockaddr_in6 *broker6 = ((struct sockaddr_in6 *)broker);
+			char ipv6_addr[NET_IPV6_ADDR_LEN];
+
+			memcpy(broker6->sin6_addr.s6_addr,
+			       ((struct sockaddr_in6 *)addr->ai_addr)->sin6_addr.s6_addr,
+			       sizeof(struct in6_addr));
+			broker6->sin6_family = AF_INET6;
+			broker6->sin6_port = htons(CONFIG_MQTT_HELPER_PORT);
+
+			inet_ntop(AF_INET6, &((struct sockaddr_in6 *)addr->ai_addr)->sin6_addr,
```

Comment:
Could use `broker6` pointer here, the address is already copied and it'd be bit simpler syntax.

Reply:
I moved the conversion earlier as I realised the inet_ntop function is based on the family passed in, so works before you convert.